src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,238 @@
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                cars.add(getCar(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get a list of cars from driversDB.", e);
+        }
+        for (Car car : cars) {
+            if (car != null) {
+                car.setDrivers(getDriversForCar(car.getId()));
+            }
+        }
+        return cars;
+    }
+
+    private List<Driver> getDriversForCar(Long carId) {
+        String getAllDriversForCarRequest = "SELECT id, name, license_number "
+                + "FROM drivers "
+                + "JOIN cars_drivers "
+                + "ON drivers.id = cars_drivers.driver_id "
+                + "WHERE cars_drivers.car_id = ?;";
+        List<Driver> drivers = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getAllDriversStatement =
+                        connection.prepareStatement(getAllDriversForCarRequest)) {
+            getAllDriversStatement.setLong(1, carId);
+            ResultSet resultSet = getAllDriversStatement.executeQuery();
+            while (resultSet.next()) {
+                drivers.add(parseDriversFromResultSet(resultSet));
+            }
+            return drivers;
+        } catch (SQLException e) {
+            throw new RuntimeException("Can't find cars in DB by driver id=" + carId, e);
+        }
+    }
+
+    private Driver parseDriversFromResultSet(ResultSet resultSet) throws SQLException {
+        Driver driver = new Driver();
+        driver.setId(resultSet.getObject("drivers.id", Long.class));
+        driver.setName(resultSet.getString("drivers.name"));
+        driver.setLicenseNumber(resultSet.getString("drivers.license_number"));
+        return driver;
+    }
+
+    @Override
+    public Car update(Car car) {
+        String query = "UPDATE cars "
+                + "SET model = ?, manufacturer_id = ? "
+                + "WHERE id = ? AND is_deleted = FALSE";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement
+                        = connection.prepareStatement(query)) {
+            statement.setString(1, car.getModel());
+            statement.setLong(2, car.getManufacturer_id());
+            statement.setLong(3, car.getId());
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't update "
+                    + car + " in driversDB.", e);
+        }
+        deleteAllDriversFromCar(car.getId());
+        insertDriversToCar(car);
+        return car;
+    }
+
+    private void deleteAllDriversFromCar(Long id) {
+        String query = "DELETE FROM cars_drivers "
+                + "WHERE car_id = ? AND driver_id = ?;";
```

Comment:
```suggestion
        String query = "DELETE FROM cars_drivers "
                + "WHERE car_id = ?";
```

I think we can just remove all drivers for a specific car like this, and remove getAllDriversIdByCar method